### PR TITLE
prov/verbs: Use only active ports

### PR DESCRIPTION
**Pull request by gladkovdmitry17**
This patch addresses the problem when the coming first ports are in the state DOWN, and after these ports there are ACTIVE ports.
Now all DOWN ports is skipping and the first found ACTIVE port is chosen

Signed-off-by: Dmitry Gladkov <dmitry.gladkov@intel.com>

### CODE REVIEW

File: prov/verbs/src/verbs_info.c [hunk 1]
```diff
@@ -518,11 +519,26 @@ static int fi_ibv_get_device_attrs(struct ibv_context *ctx,
...
+					 "ibv_query_port", errno);
+			return -errno;
+		}
+		if (port_attr.state == IBV_PORT_ACTIVE) {
+			port_num = i;
+			break;
+		}
+	}
+
+	if (!port_num) {
```
Comment (shefty) [inline prov/verbs/src/verbs_info.c]: nit: you can remove port_num, and compare i == 32, with any other value i is the port num

Comment (gladkovdmitry17) [reply prov/verbs/src/verbs_info.c]: Thanks, good catch!

Comment (gladkovdmitry17) [reply prov/verbs/src/verbs_info.c]: @shefty Not sure why, but the changes from [my fork](https://github.com/gladkovdmitry17/libfabric/commit/c09924b361bd4edc4fc3c2e9094a431befe05c8b) isn't synced with the PR

Comment (gladkovdmitry17) [reply prov/verbs/src/verbs_info.c]: @shefty Fixed. Now it works :)

**Comment by shefty** [comment]
Looks good -- thanks

**Comment by gladkovdmitry17** [comment]
@shefty @a-ilango Thanks for the review!

**Pull request merged**